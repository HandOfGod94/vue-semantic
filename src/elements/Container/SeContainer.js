import SeContainerComponent from './SeContainer.functional'

export default {
  name: 'se-container',

  props: {
    text: Boolean,
    align: String,
    justified: Boolean,
    fluid: Boolean
  },

  computed: {
    classObject () {
      return {
        ui: true,
        text: this.text,
        [`${this.align} aligned`]: this.align,
        justified: this.justified,
        fluid: this.fluid,
        container: true
      }
    }
  },

  render (createElement) {
    return createElement(SeContainerComponent, {
      attrs: {
        slots: this.$slots
      },
      class: this.classObject
    })
  }
}
