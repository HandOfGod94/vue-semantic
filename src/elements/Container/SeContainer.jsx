import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-container',
  props: {
    text: Boolean,
    align: String,
    justified: Boolean,
    fluid: Boolean
  }
})
class SeContainer extends Vue {
  render () {
    return (
      <div class={this.classes}>
        {this.$slots.default}
      </div>
    )
  }

  get classes () {
    const classes = {
      ui: true,
      text: this.text,
      [`${this.align} aligned`]: this.align,
      justified: this.justified,
      fluid: this.fluid,
      container: true
    }

    return classes
  }
}

export default SeContainer
