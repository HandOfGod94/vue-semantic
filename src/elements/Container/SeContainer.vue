<template>
  <div :class="classObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'se-container',
  props: {
    text: Boolean,
    align: String,
    justified: Boolean,
    fluid: Boolean
  },
  computed: {
    classObject () {
      return {
        ui: true,
        text: this.text,
        [`${this.align} aligned`]: this.align,
        justified: this.justified,
        fluid: this.fluid,
        container: true
      }
    }
  }
}
</script>
