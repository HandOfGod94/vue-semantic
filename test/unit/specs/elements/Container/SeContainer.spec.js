import SeContainer from '@/elements/Container/SeContainer'
import { mount } from '@vue/test-utils'

const contentWrapper = `
  <div> Hello World </div>
`

describe('SeContainer', () => {
  it('should have margins as per text attr', () => {
    const data = { propsData: { text: true }, slots: { default: contentWrapper } }
    const wrapper = mount(SeContainer, data)
    expect(wrapper.contains('.text')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should align properly', () => {
    const data = { propsData: { align: 'left' }, slots: { default: contentWrapper } }
    const wrapper = mount(SeContainer, data)
    expect(wrapper.classes()).toContain('left')
    expect(wrapper.classes()).toContain('aligned')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should contiain class for justified', () => {
    const data = { propsData: { justified: true }, slots: { default: contentWrapper } }
    const wrapper = mount(SeContainer, data)
    expect(wrapper.classes()).toContain('justified')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should contiain class for fluid', () => {
    const data = { propsData: { fluid: true }, slots: { default: contentWrapper } }
    const wrapper = mount(SeContainer, data)
    expect(wrapper.classes()).toContain('fluid')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
