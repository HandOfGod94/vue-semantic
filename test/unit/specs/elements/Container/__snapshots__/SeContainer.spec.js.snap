// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeContainer should align properly 1`] = `
<div class="ui left aligned container">
    <div> Hello World </div>
</div>
`;

exports[`SeContainer should contiain class for fluid 1`] = `
<div class="ui fluid container">
    <div> Hello World </div>
</div>
`;

exports[`SeContainer should contiain class for justified 1`] = `
<div class="ui justified container">
    <div> Hello World </div>
</div>
`;

exports[`SeContainer should have margins as per text attr 1`] = `
<div class="ui text container">
    <div> Hello World </div>
</div>
`;
